### PR TITLE
chore(deps): update `cross-spawn` to v7.0.6

### PR DESCRIPTION
**Pull request by dargmuesli**
## What's the problem this PR addresses?

There's a  Regular Expression Denial of Service (ReDoS) in cross-spawn >= 7.0.0, < 7.0.5; < 6.0.6

## How did you fix it?

Update `cross-spawn` to v7.0.6, including all transitive dependencies

## Checklist

- [x] I have read the [Contributing Guide](https://yarnpkg.com/advanced/contributing).

- [x] I have set the packages that need to be released for my changes to be effective.

- [x] I will check that all automated PR checks pass before the PR gets reviewed.

### CODE REVIEW

File: .yarn/versions/2f387047.yml [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+  "@yarnpkg/plugin-patch": patch
+  "@yarnpkg/plugin-pnp": patch
+  "@yarnpkg/plugin-pnpm": patch
+  "@yarnpkg/plugin-stage": patch
+  "@yarnpkg/plugin-typescript": patch
+  "@yarnpkg/plugin-version": patch
+  "@yarnpkg/plugin-workspace-tools": patch
+  "@yarnpkg/pnpify": patch
+  "@yarnpkg/sdks": patch
+  "@yarnpkg/shell": patch
```
Comment (merceyz) [inline .yarn/versions/2f387047.yml:33]: Only need a patch release on core, shell, and the cli.

**Review by merceyz — CHANGES_REQUESTED** [review]
Based on https://github.com/moxystudio/node-cross-spawn/pull/160 this seems unlikely to be an issue we should be concerned about.

> including all transitive dependencies

That's unnecessary.

File: .yarn/versions/2f387047.yml (hunk 1 above)
Comment (dargmuesli) [reply .yarn/versions/2f387047.yml]: Updated :white_check_mark:

**Comment by dargmuesli** [comment]
> That's unnecessary.

I'd argue it's necessary as without updating `cross-spawn` in all places, the vulnerability could be triggered in a transitive dependency. Although I agree this is unlikely. It was just appeared doable to update the dependency, so I thought I propose this change.

**Comment by merceyz** [comment]
Thanks for the PR but I'll close this since it isn't a vulnerability we need to worry about and with https://github.com/yarnpkg/berry/pull/6606 merged and released consumers of the npm packages will get the patched version of cross-spawn.

**Pull request closed without merging**